### ISSUE TITLE
Bootstrap 4

### ISSUE THREAD

**Issue by metanav** [zebzhao/Angular-QueryBuilder#24]
I am using Bootstrap 4 in my application. When I apply 'classNames' the querybuilder view is not rendered as expected.

**Comment by zebzhao** [comment]
Hi, can you post an example what of you mean? I haven't tested with Bootstrap 4.

**Comment by metanav** [comment]
You can see in the rendered UI below the fields are on different  rows and rule/ruleset buttons are aligned to left.

<img width="892" alt="screen shot 2018-03-07 at 11 40 58" src="https://user-images.githubusercontent.com/7456984/37070765-8af2aff2-21fc-11e8-89ee-eb33d705fbec.png">

The configs I am using:
```
  query = {
    condition: 'and',
    rules: [
      {field: 'value', operator: 'equal'}
    ]
  };

  config: any = {
    fields: {
      key: {name: 'Key', type: 'key', operators: ['equal']},
      value: {name: 'Value', type: 'value', operators: ['equal']},
    }
  };

  classNames: {[key: string]: string} = {
    removeIcon: 'fa fa-minus',               // Font-Awesome
    addIcon: 'fa fa-plus',                   // Font-Awesome
    button: 'btn',                           // Bootstrap
    buttonGroup: 'btn-group',                // Bootstrap
    switchGroup: 'btn-group',                // Bootstrap
    queryTree: 'q-tree',                     // DEFAULT
    queryItem: 'q-item',                     // DEFAULT
    queryRule: 'q-rule',                     // DEFAULT
    queryRuleSet: 'q-ruleset',               // DEFAULT
    invalidRuleset: 'alert alert-danger',    // Bootstrap
    emptyWarning: '',                        // Bootstrap
    fieldControl: 'form-control',            // Bootstrap
    operatorControl: 'form-control',         // Bootstrap
    inputControl: 'form-control'             // Bootstrap
  };
```

**Comment by zebzhao** [comment]
I realize that the CSS classes are not specific enough for bootstrap to properly work. I also found a bug regarding nested Ruleset not having classNames applied properly. I'll fix this in the next release of the library, although there may be additional classNames properties to accommodate for finer control of styling with 3rd party libraries.

**Comment by zebzhao** [comment]
Released version 0.2.4 which has support for bootstrap 4.
